domn1995/dunet
# Separate UnionAttribute into its own project

fixes #147

- Splits the library into 2 projects, the attribute and the generator
- Moves common project items to the Directory.Build.props
- Updates Benchmarks to remove covering code that is no longer generated
- Updates tests to work with new project layout
- Adds SDK reference to PackAsAnalyzer
- Updates namespaces

## Testing

To test this locally, use the following steps:

- Create a folder in the project root called `.packages`
- Create a file in the project root named `NuGet.Config` with the following contents:

```xml
<configuration>
  <packageSources>
    <add key="local" value="./.packages" />
  </packageSources>
</configuration>
```

- Open a command window in the project root and run `dotnet build -c Release --no-restore`
- Copy the output nupkg file from `src\Dunet\bin\Release` to the `.packages` folder
- Update integration tests to use Dunet 1.11.0, they should still pass, and control clicking on the attribute should (IDE dependent) take you to the decompiled attribute source.
